### PR TITLE
tools/testing: update dist-check to use rockylinux and adapt to cmake

### PR DESCRIPTION
**Pull request by tchaikov**
`dist-check` tests the generated rpm packages by installing them in a centos 7 container. but this script is terribly outdated

- centos 7 is deprecated. we should use a new distro's latest stable release.
- cqlsh was added to the family of rpms a while ago. we should test it as well.
- the directory hierarchy has been changed. we should read the artifacts from the new directories. 
- cmake uses a different directory hierarchy. we should check the directory used by cmake as well.

to address these breaking changes, the scripts are updated accordingly.

---

this change gives an overhaul to a test, which is not used in production. so no need to backport.

### CODE REVIEW

**Comment by regevran** [comment]
Can you add a test that checks `dist-test` - making sure it is not going to lose sync ever again?

File: tools/testing/dist-check/dist-check.sh [hunk 1]
```diff
@@ -51,7 +51,7 @@ if [ -f /.dockerenv ]; then
 fi
 
 container_images=(
-    docker.io/centos:7
+    docker.io/fedora:latest
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply tools/testing/dist-check/dist-check.sh]: ack. changed to rockylinux-9.

**Comment by tchaikov** [comment]
> Can you add a test that checks `dist-test` - making sure it is not going to lose sync ever again?

that's the plan. but i don't want to mingle changes with different purposes into a single one, the goal is to limit the scope, and help the reviewer and contributor to focus on one thing and only one thing.

the scope of this change, as its title claims is to *fix* "dist-check" to use a stable release of another RPM based distro, and update the scripts to cater the breaking changes happened in the past years.

to "add" the test is not in the scope. but #20266 prepares for it. and please note neither does #20266 is supposed to *add* the test, as adding another test is a different story, and needs a different venue for another discussion.

**Comment by tchaikov** [comment]
v2:

- use rockylinux:9 for testing
- move the changes updating the paths in `docker.io/util.sh` to the proper commit. so that the last commit of "tools/testing: add cqlsh rpm" only contains the change adding the cqlsh rpm package.

**Comment by tchaikov** [comment]
> #### 🔴 CI State: FAILURE
> 
> ❌ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11105/)
> #### Build Details:
> 
>     * Duration: 35 min
> 
>     * Builder: spider5.cloudius-systems.com

```
[2024-08-23T08:44:22.105Z] ++ buildah from docker.io/ubuntu:24.04
[2024-08-23T08:44:22.105Z] Error during unshare(...): Operation not permitted
...
[2024-08-23T08:44:24.983Z] + podman pull
[2024-08-23T08:44:24.983Z] Error: requires at least 1 arg(s), only received 0
[2024-08-23T08:44:25.437Z] [2682/3261] LINK build/debug/test/boost/memtable_test
[2024-08-23T08:44:25.584Z] [Pipeline] echo
[2024-08-23T08:44:25.592Z] ERROR: container build (x86_64) failed with error: script returned exit code 125
```
infra issues.

**Comment by regevran** [comment]
> > Can you add a test that checks `dist-test` - making sure it is not going to lose sync ever again?
> 
> the scope of this change, as its title claims is to _fix_ "dist-check" to use a stable release of another RPM based distro, and update the scripts to cater the breaking changes happened in the past years.
> 
> to _add_ the test is not in the scope of this change. but #20266 prepares for adding the test. and please note neither does #20266 is supposed to _add_ the test, as adding another test is a different story, and needs a different venue for another discussion.

Sorry, I find it hard to follow the logic.
This PR fixes an outdated code. The code got outdated because no one needed it (no complains about it being broken) and no one tested it - no automated test for this. 
This what it does: 
> dist-check tests the generated rpm packages by installing them in a centos 7 container.

And as said - there is no code that verifies that this installation actually works.
It means that the supplied fix will get outdated in the future, because the underlying reason for it to get outdated was not solved. Specifically - there is no code that tests it.

> that's the plan. but i don't want to mingle changes with different purposes into a single one. the goal is to narrow the focus, enabling reviewers and contributors to concentrate on a single, specific aspect at a time. this approach enhances clarity and efficiency in the review process.
> 
It is OK to split into two tasks, but the other task should be opened. There is no issue or pr for adding (modifying existing?) the required test.

**Comment by tchaikov** [comment]
@avikivity could you please take another look?

**Comment by tchaikov** [comment]
> > > Can you add a test that checks `dist-test` - making sure it is not going to lose sync ever again?
> > 
> > 
> > the scope of this change, as its title claims is to _fix_ "dist-check" to use a stable release of another RPM based distro, and update the scripts to cater the breaking changes happened in the past years.
> > to _add_ the test is not in the scope of this change. but #20266 prepares for adding the test. and please note neither does #20266 is supposed to _add_ the test, as adding another test is a different story, and needs a different venue for another discussion.
> 
> Sorry, I find it hard to follow the logic. This PR fixes an outdated code. The code got outdated because no one needed it (no complains about it being broken) and no one tested it - no automated test for this. This what it does:
> 
> > dist-check tests the generated rpm packages by installing them in a centos 7 container.
> 
> And as said - there is no code that verifies that this installation actually works. It means that the supplied fix will get outdated in the future, because the underlying reason for it to get outdated was not solved. Specifically - there is no code that tests it.
> 
> > that's the plan. but i don't want to mingle changes with different purposes into a single one. the goal is to narrow the focus, enabling reviewers and contributors to concentrate on a single, specific aspect at a time. this approach enhances clarity and efficiency in the review process.
> 
> It is OK to split into two tasks, but the other task should be opened. There is no issue or pr for adding (modifying existing?) the required test.

discussed with Ran offline. and i explained why i prefer not creating an issue tracking this effort. and i do have a PR for preparing for the follow-up changes.
